src/components/TaskListContainer.tsx

```diff
@@ -0,0 +1,30 @@
+import { Dispatch, SetStateAction } from 'react';
+import { Task } from '../utils/types';
+import TaskListUI from './TaskListUI';
+
+interface TaskListContainerProps {
+  tasks: Task[];
+  setTasks: Dispatch<SetStateAction<Task[]>>;
+}
+
+function TaskListContainer({ tasks, setTasks }: TaskListContainerProps) {
+  const uncompletedTasks = tasks.filter((task) => task.completed === false); //filter for uncompleted tasks
+  const completedTasks = tasks
+    .filter((task) => task.completed === true) //filter for completed tasks
+    .sort(
+      (a, b) =>
+        new Date(b.updatedAt).getTime() - new Date(a.updatedAt).getTime(), //sort filtered tasks by updatedAt time
```

Comment:
Question 🤔: what date format are you saving these as in your database? Are these ISO? Does locale matter?

Reply:
the dates are in ISO format. I don't think locale matters as the dates/times are not being displayed in the app (wasn't a requirement originally) and are only being used for sorting the tasks by time.

Also added a helper function in the component called `fiterTasksByCompletionStatus` to filter between completed/uncompleted tasks consistently.